azavea/raster-foundry
# Remove resource-specific metrics

## Overview

This removes per-resource metrics from Kamon. Kamon automatically generates metrics for any interaction with an outside service (in our case, usually Rollbar). This seems like a reasonable thing to do because it gives us information about how much latency outside service calls are contributing to our overall latency.

However, when combined with the fact that we were also generating metrics for individual resources (projects, layers, and tools), it was causing our metric count to blow up because each resource would spawn several metrics if it ever encountered an error which was logged to Rollbar.

This removes all metrics which are generated on a per-resource basis.

### Checklist


### Notes

I tried to use Kamon's `filter` configuration settings to filter out the Rollbar metrics so that we could keep some per-resource metrics around, but I wasn't able to determine the correct incantation to convince Kamon to drop those metrics.

I also looked into disabling the automatic metrics for external service calls, but there doesn't seem to be a way to disable just those metrics in the `kamon-akka-http` configuration. Based on inspecting the `kamon-akka-http` source, I think that setting a `host-level` instrumentation level would cause these service calls to not be instrumented, but that's because `kamon-akka-http` doesn't support `host-level` instrumentation yet and returns an empty trace if you try it. This didn't seem like a very good solution since `kamon-akka-http` will presumably add support for this setting in the future which could cause the metrics to return unexpectedly.

The boot time for the server has gotten pretty long, which made it difficult to iterate quickly on different configurations.

Because of this, I decided to start by removing resource ids from all `traceName` calls so that those calls would only return a single trace rather than a separate trace for each resource. This should reduce the number of metrics to a manageable level, although we may want to add some of the per-resource metrics back in later if we start running into problems with particular projects, layers, or tools.

## Testing Instructions

 * Bring up your server by running `./scripts/server --with-metrics`
 * Once the server has booted, load the Projects list page (probably twice to get thumbnails).
 * statsd should only log about 60 for `numStats` after doing this (it was previously 150+ depending on the number of projects).
 * Follow the directions in https://github.com/azavea/raster-foundry/pull/2120 to get Grafana configured to pull from Graphite.
 * Open Grafana by visiting localhost:3000, and add a new graph. Edit it to add a different query and select `stats->development->tile->counter` and note that none of the `LayerCache_*` and `Mosaic_*` metrics have UUIDs associated.

Closes #2177
